src/main/java/com/owncloud/android/lib/common/OwnCloudClient.java

```diff
@@ -199,6 +204,10 @@ public int executeMethod(HttpMethod method) throws IOException {
 
             int status = super.executeMethod(method);
 
+            if (status >= 500 && status < 600 && DNSCache.INSTANCE.isIPV6(hostname)) {
```

Comment:
Have you tried this also with e.g. http://10.0.2.2/nc or http:/${ipv6 address}

Reply:
Works fine with IPv4.

With IPv6, it gets stuck trying to do an HTTPS connection (and I have no local HTTPS) instead of trying just HTTP. Is this a separate bug, or did it use to work?

Reply:
Fixed by https://github.com/nextcloud/android-library/pull/793/commits/29c5eda7831602d861640e699f9ff8f31e6ef34d. Good catch!